[2023-12-19T11:56:40.049+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_dag.run_spark_script scheduled__2023-12-11T00:12:00+00:00 [queued]>
[2023-12-19T11:56:40.060+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_dag.run_spark_script scheduled__2023-12-11T00:12:00+00:00 [queued]>
[2023-12-19T11:56:40.061+0000] {taskinstance.py:2171} INFO - Starting attempt 1 of 6
[2023-12-19T11:56:40.123+0000] {taskinstance.py:2192} INFO - Executing <Task(SparkSubmitOperator): run_spark_script> on 2023-12-11 00:12:00+00:00
[2023-12-19T11:56:40.141+0000] {standard_task_runner.py:60} INFO - Started process 1518 to run task
[2023-12-19T11:56:40.145+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'spark_dag', 'run_spark_script', 'scheduled__2023-12-11T00:12:00+00:00', '--job-id', '131', '--raw', '--subdir', 'DAGS_FOLDER/spark_job_dag.py', '--cfg-path', '/tmp/tmpg685vw35']
[2023-12-19T11:56:40.147+0000] {standard_task_runner.py:88} INFO - Job 131: Subtask run_spark_script
[2023-12-19T11:56:40.283+0000] {task_command.py:423} INFO - Running <TaskInstance: spark_dag.run_spark_script scheduled__2023-12-11T00:12:00+00:00 [running]> on host ea216ca38749
[2023-12-19T11:56:40.397+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='limoo' AIRFLOW_CTX_DAG_ID='spark_dag' AIRFLOW_CTX_TASK_ID='run_spark_script' AIRFLOW_CTX_EXECUTION_DATE='2023-12-11T00:12:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-12-11T00:12:00+00:00'
[2023-12-19T11:56:40.406+0000] {base.py:83} INFO - Using connection ID 'spark_default' for task execution.
[2023-12-19T11:56:40.407+0000] {spark_submit.py:351} INFO - Spark-Submit cmd: spark-submit --master spark://master:4040 --name arrow-spark ./spark_job.py
[2023-12-19T11:56:40.470+0000] {spark_submit.py:521} INFO - /home/***/.local/lib/python3.8/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2023-12-19T11:56:40.474+0000] {spark_submit.py:521} INFO - JAVA_HOME is not set
[2023-12-19T11:56:40.498+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 160, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 452, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://master:4040 --name arrow-spark ./spark_job.py. Error code is: 1.
[2023-12-19T11:56:40.501+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_dag, task_id=run_spark_script, execution_date=20231211T001200, start_date=20231219T115640, end_date=20231219T115640
[2023-12-19T11:56:40.541+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 131 for task run_spark_script (Cannot execute: spark-submit --master spark://master:4040 --name arrow-spark ./spark_job.py. Error code is: 1.; 1518)
[2023-12-19T11:56:40.598+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2023-12-19T11:56:40.651+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
